src/main/java/io/sharedstreets/tools/builder/model/BaseSegment.java

```diff
@@ -49,6 +49,20 @@ public static boolean canMerge(BaseSegment baseSegment1, BaseSegment baseSegment
                 return false;
         }
 
+	// can't merge primary road classes with other road classes
+	//
+	// Motivation:
+	// We're merging aggressively, and creating mixed roadClass references to prevent poorly mapped OSM ways getting
+	// split up when they should be a contiguous reference. However, primary roads are often mapped with better quality
+	// and it makes sense to handle start/end of motorway/trunk roads properly.
+
+	int baseSegment1RoadClass = baseSegment1.getRoadClass().getValue();
+        int baseSegment2RoadClass = baseSegment2.getRoadClass().getValue();
+	int roadClassMergeThreshold = Way.ROAD_CLASS.ClassPrimary.getValue();
+        if(baseSegment1RoadClass != baseSegment2RoadClass && (baseSegment1RoadClass <= roadClassMergeThreshold || baseSegment2RoadClass <= roadClassMergeThreshold))
```

Comment:
What if both are below the threshold? I think in this case they should be mergeable, but I don't think they are with this logic.

Reply:
I'd suggest:
```java
if(baseSegment1RoadClass != baseSegment2RoadClass) {
  if ((baseSegment1RoadClass < roadClassMergeThreshold && baseSegment2RoadClass >= roadClassMergeThreshold) || 
      (baseSegment2RoadClass < roadClassMergeThreshold && baseSegment1RoadClass >= roadClassMergeThreshold)) {
    return false;
  }
}
```